lib/internal/async_hooks.js

```diff
@@ -360,6 +377,9 @@ function clearAsyncIdStack() {
   async_id_fields[kExecutionAsyncId] = 0;
   async_id_fields[kTriggerAsyncId] = 0;
   async_hook_fields[kStackLength] = 0;
+  while (execution_async_resources.length) {
+    execution_async_resources.shift();
+  }
```

Comment:
The fastest way to make an array empty in recent V8 versions: `execution_async_resources.splice(0, execution_async_resources.length)`.